deepchecks/tabular/context.py

```diff
@@ -30,16 +33,118 @@
 ]
 
 
+class _DummyModel:
+    """Contains all the data + properties the user has passed to a check/suite, and validates it seamlessly.
+
+    Parameters
+    ----------
+    train: Dataset
+        Dataset, representing data an estimator was fitted on
+    test: Dataset
+        Dataset, representing data an estimator predicts on
+    y_pred_train: np.ndarray
+        Array of the model prediction over the train dataset.
+    y_pred_test: np.ndarray
+        Array of the model prediction over the test dataset.
+    y_proba_train: np.ndarray
+        Array of the model prediction probabilities over the train dataset.
+    y_proba_test: np.ndarray
+        Array of the model prediction probabilities over the test dataset.
+    """
+
+    train: pd.DataFrame
+    test: pd.DataFrame
+    ind_map: t.Dict[str, t.Dict[t.Any, int]]
+    y_pred: t.Dict[str, np.ndarray]
+    y_proba: t.Dict[str, np.ndarray]
+
+    def __init__(self,
+                 train: Dataset,
+                 test: Dataset,
+                 y_pred_train: np.ndarray,
+                 y_pred_test: np.ndarray,
+                 y_proba_train: np.ndarray,
+                 y_proba_test: np.ndarray,):
+        self.ind_map = defaultdict(dict)
+
+        if train is not None and test is not None:
+            # check if datasets have same indexes
+            if set(train.data.index) & set(test.data.index):
+                train.data.index = map(lambda x: f'train-{x}', list(train.data.index))
+                test.data.index = map(lambda x: f'test-{x}', list(test.data.index))
+                warnings.warn('train and test datasets have common index - adding "train"/"test"'
+                              ' prefixes. To avoid that provide datasets with no common indexes '
+                              'or don\'t use the static predictions feature.')
+
+        if train is not None:
+            self.train = train.features_columns
+            for i, ind in enumerate(self.train.index):
+                self.ind_map[DatasetKind.TRAIN][ind] = i
+
+        if test is not None:
+            self.test = test.features_columns
+            for i, ind in enumerate(self.test.index):
+                self.ind_map[DatasetKind.TEST][ind] = i
```

Comment:
why do you need to separate the map to train & test if the indices are mutually exclusive?

Reply:
because I need to know which indices belong to where to pick from the right prediction

Reply:
you are anyway mapping the indices why not just have a map of index to prediction?

Reply:
```
for index, prediction in zip(test.index, test_predictions):
   self.pred_map[index] = prediction
```

Reply:
but then I can't validate the data

Reply:
also it will make the prediction to be of type dataframe instead of numpy array (or like need to be aggregated)

Reply:
what does it mean validate the data

Reply:
and dataframe and numpy are very easy to convert between each other

Reply:
it is but it makes the data preparation + returning harder
and i can't validate if the data is from the right dataset if I don't know which dataset it is